tracer/src/Datadog.Trace/Sampling/ISampler.cs

```diff
@@ -9,7 +9,7 @@ namespace Datadog.Trace.Sampling
 {
     internal interface ISampler
     {
-        void SetDefaultSampleRates(IEnumerable<KeyValuePair<string, float>> sampleRates);
+        void SetDefaultSampleRates(IReadOnlyDictionary<string, float> sampleRates);
```

Comment:
`Dictionary<string, float>` implements `IEnumerable<KeyValuePair<string, float>>` so I don't think this change was technically required. Was there reason for it? Without looking too deep, it seems to me like `IEnumerable` is more flexible (we could use `Dictionary` or `List` or arrays internally, whatever fits the situation best).

Reply:
I removed it because I used to do a double enumeration at some point to get the default samples, and wanted to avoid that. Now the only small win is to create a dictionary of the right size directly. 
That said, I honestly I don't see the value of relying on IEnumerable if anyway we have underlying dictionaries in all cases.

Reply:
Ok, thanks.